### PR TITLE
Update analyzers

### PR DESCRIPTION
**Pull request by pakrym**
More new rules:
Uppercase tuple names
Use tuple names
Space after target-typed new

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Diagnostics/BlobsCheckpointStoreDiagnosticsTests.cs [hunk 1]
```diff
@@ -613,7 +613,7 @@ private class MockBlobContainerClient : BlobContainerClient
             public override string Name { get; }
             internal IEnumerable<BlobItem> Blobs;
             internal Exception GetBlobsAsyncException;
-            internal Dictionary<string, MockBlobClient> BlobClients = new ();
+            internal Dictionary<string, MockBlobClient> BlobClients = new();
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Diagnostics/BlobsCheckpointStoreDiagnosticsTests.cs:616]: yay!

**Review by jsquire — APPROVED** [review]
Event Hubs and Service Bus bits look good to me.

**Review by christothes — APPROVED** [review]

File: sdk/search/Azure.Search.Documents/src/Batching/SearchIndexingPublisher.cs [hunk 2]
```diff
@@ -235,7 +235,7 @@ protected override async Task<bool> OnSubmitBatchAsync(IList<PublisherAction<Ind
         /// <param name="actions">The batch of actions.</param>
         /// <param name="results">The results.</param>
         /// <returns>Actions paired with their result.</returns>
-        private static IEnumerable<(PublisherAction<IndexDocumentsAction<T>>, IndexingResult)> AssociateResults(
+        private static IEnumerable<(PublisherAction<IndexDocumentsAction<T>> Action, IndexingResult Result)> AssociateResults(
```
Comment (heaths) [inline sdk/search/Azure.Search.Documents/src/Batching/SearchIndexingPublisher.cs:238]: I'm slightly concerned I don't see other changes that reference these values. @tg-msft can you take a look?

File: sdk/search/Azure.Search.Documents/src/Indexes/FieldBuilder.cs [hunk 3]
```diff
@@ -229,7 +229,7 @@ ArgumentException FailOnUnknownDataType()
             return info.Properties
                 .Select(prop => (prop.Name, BuildField(prop)))
                 .Where(pair => pair.Item2 != null)
-                .ToDictionary(pair => pair.Item1, pair => pair.Item2);
+                .ToDictionary(pair => pair.Name, pair => pair.Item2);
```
Comment (heaths) [inline sdk/search/Azure.Search.Documents/src/Indexes/FieldBuilder.cs:232]: Why wasn't `Item2` updated?

**Review by heaths — COMMENTED** [review]
I have slight concerns with Search changes (and might be a problem elsewhere), but, also, how are these new rules getting pulled in? Besides StyleCop.Analyzers, I don't see any source or reference. Or are all these new rules from there - i.e. not custom AZC ones?

File: sdk/search/Azure.Search.Documents/src/Indexes/FieldBuilder.cs (hunk 3 above)
Comment (pakrym) [reply sdk/search/Azure.Search.Documents/src/Indexes/FieldBuilder.cs:232]: It's not required to name tuple fields when they are not part of the API surface but it's required that you use the named field when it exists.

File: sdk/search/Azure.Search.Documents/src/Batching/SearchIndexingPublisher.cs (hunk 2 above)
Comment (pakrym) [reply sdk/search/Azure.Search.Documents/src/Batching/SearchIndexingPublisher.cs:238]: I suspect deconstruction was used at the consumption points.

**Comment by pakrym** [comment]
>how are these new rules getting pulled in? Besides StyleCop.Analyzers, I don't see any source or reference

They are part of StyleCop.

File: sdk/search/Azure.Search.Documents/src/Batching/SearchIndexingPublisher.cs (hunk 2 above)
Comment (tg-msft) [reply sdk/search/Azure.Search.Documents/src/Batching/SearchIndexingPublisher.cs:238]: Yes, we just deconstruct at https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/search/Azure.Search.Documents/src/Batching/SearchIndexingPublisher.cs#L169

**Review by kinelski — APPROVED** [review]
Text Analytics and Metrics Advisor LGTM

File: sdk/search/Azure.Search.Documents/src/Indexes/FieldBuilder.cs (hunk 3 above)
Comment (heaths) [reply sdk/search/Azure.Search.Documents/src/Indexes/FieldBuilder.cs:232]: Right - IIRC, we shouldn't use tuples in public APIs anyway. But `Item2` has a name and it wasn't used here.

**Pull request merged**